sdks/python/apache_beam/typehints/typecheck.py

```diff
@@ -87,14 +92,14 @@ def wrapper(self, method, args, kwargs):
     except TypeCheckError as e:
       error_msg = ('Runtime type violation detected within ParDo(%s): '
                    '%s' % (self.full_label, e))
-      six.raise_from(TypeCheckError(error_msg), sys.exc_info()[2])
+      raise_with_traceback(TypeCheckError(error_msg))
     else:
       return self._check_type(result)
 
   def _check_type(self, output):
     if output is None:
       return output
-    elif isinstance(output, (dict,) + six.string_types):
+    elif isinstance(output, (dict, str, unicode)):
```

Comment:
We should add `bytes` to this tuple: this preserves the intended behavior in Python 3, where `str` and `unicode` here are synonyms.

Reply:
I don't think we need to add `bytes` here.
`six.string_types` is equivalent to `basestring` on python 2 and `str` on python 3. `(str, unicode)` is equivalent to `basestring` on Python 2 and here also equivalent to `str` on Python 3 due to the `unicode=str` assignment.

Reply:
My understanding is that in Python 2, `bytes` is equivalent to `str`, but this is not the case in Python 3, so by leaving out `bytes` in Python 3, this code path would not be triggered if `output` is of type `bytes`, so that we should add this to the list.  Is that correct?

Reply:
You're right. We also want to trigger this path for `bytes`.
I'll add it.